Plugins/NativeEngine/Source/NativeDataStream.h

```diff
@@ -0,0 +1,223 @@
+#pragma once
+
+#include <Babylon/JsRuntime.h>
+
+#include <napi/env.h>
+
+#include <gsl/gsl>
+
+namespace Babylon
+{
+    class NativeDataStream final : public Napi::ObjectWrap<NativeDataStream>
+    {
+        static constexpr auto JS_CLASS_NAME = "_NativeDataStream";
+        static constexpr auto JS_ENGINE_CONSTRUCTOR_NAME = "NativeDataStream";
+
+        static constexpr auto VALIDATION_ENABLED = false;
```

Comment:
is this in sync with the JS side somehow?

Reply:
The first two lines are "in sync" using the same pattern as `NativeEngine`; nobody needs to know the first, and the second is hard-coded in the relevant TS file. The validation thing right now must be changed manually on both sides. That was done that way with the goal of making it able to completely disappear on both sides if it's not being used (compiled out in C++ and not even included in TS), but the pattern might be worth discussing more broadly in case people have better ideas.

Reply:
I'm only making a comment on the VALIDATION_ENABLED flag. I would expect the JS side to read this flag on the native side so that they are in sync. At least check that both sides are enabled.

Reply:
The way I did it in the other PR, there was some minimal extra code executed even when validation was enabled, but the difference was negligible when I profiled it. I had considered doing something fancier to have no extra code, but it seemed like unnecessary extra complexity. It would be nice if validation was automatically enabled by just using the `ValidatedNativeDataStream` class on the JS side.